candle-binding/src/classifiers/lora/intent_lora.rs

```
@@ -0,0 +1,138 @@
+        let classifier = HighPerformanceBertClassifier::new(model_path, num_classes, use_cpu)
+            .map_err(|e| {
+                let unified_err = model_error!(
+                    ModelErrorType::LoRA,
+                    "intent classifier creation",
+                    format!("Failed to create BERT classifier: {}", e),
+                    model_path
+                );
+                candle_core::Error::from(unified_err)
+            })?;
+
+        Ok(Self {
+            bert_classifier: classifier,
+            confidence_threshold: 0.7,
+            intent_labels,
+            model_path: model_path.to_string(),
+        })
+    }
+
+    /// Load intent labels from model config.json using unified config loader
+    fn load_labels_from_config(model_path: &str) -> Result<Vec<String>> {
+        use crate::core::config_loader;
+
+        match config_loader::load_intent_labels(model_path) {
+            Ok(result) => Ok(result),
+            Err(unified_err) => Err(candle_core::Error::from(unified_err)),
+        }
+    }
+
+    /// Classify intent using real model inference
+    pub fn classify_intent(&self, text: &str) -> Result<IntentResult> {
+        let start_time = Instant::now();
+
+        // Use real BERT model for classification
+        let (predicted_class, confidence) =
+            self.bert_classifier.classify_text(text).map_err(|e| {
+                let unified_err = model_error!(
+                    ModelErrorType::LoRA,
+                    "intent classification",
+                    format!("Classification failed: {}", e),
+                    text
+                );
+                candle_core::Error::from(unified_err)
+            })?;
+
+        // Map class index to intent label
+        let intent = if predicted_class < self.intent_labels.len() {
+            self.intent_labels[predicted_class].clone()
+        } else {
+            format!("UNKNOWN_{}", predicted_class)
```

should just fail if no class found?

---

Yes, it should fail when no class is found rather than returning a default value. This ensures the caller knows that classification failed.